lib/json_api_controller/precondition_check.rb

```diff
@@ -21,11 +21,17 @@ def precondition
 
     def precondition_fails?
       query = resource_class.where(id: resource_ids)
-      !(gen_etag(query) == precondition)
+      current_etag = gen_etag(query)
+      current_etag = "W/#{current_etag}" if weak_etag?(precondition)
+      !(current_etag == precondition)
```

Comment:
The format for the weak etags is `W/"31877a31e40bd799084519e302a5aac3"` vs `"31877a31e40bd799084519e302a5aac3"` -- I'm probably being overly cautious but are we sure it'll handle the quotes correctly?